samples/CppArExample/CppArExample.cpp

```diff
@@ -0,0 +1,353 @@
+
+  emit arcGISArViewChanged();
+}
+
+SceneQuickView* CppArExample::sceneView() const
+{
+  return m_sceneView;
+}
+
+// Set the view (created in QML)
+void CppArExample::setSceneView(SceneQuickView* sceneView)
+{
+  if (!sceneView || sceneView == m_sceneView)
+    return;
+
+  m_sceneView = sceneView;
+  emit sceneViewChanged();
+}
+
+// properties for debug mode
+void CppArExample::showPointCloud(bool visible)
+{
+  if (visible)
+    m_arcGISArView->setPointCloudColor(QColor(50, 50, 255));
+  else
+    m_arcGISArView->setPointCloudColor(QColor());
+}
+
+void CppArExample::showPlanes(bool visible)
+{
+  if (visible)
+    m_arcGISArView->setPlaneColor(QColor(255, 0, 0, 10 ));
+  else
+    m_arcGISArView->setPlaneColor(QColor());
+}
+
+// Creates an empty scene with an elevation source.
+// Mode: Full-Scale AR
+void CppArExample::createEmptyScene()
+{
+  // create an empty scene with elevation
+  m_scene = new Scene(this);
+  createSurfaceWithElevation();
+
+  // Set the location data source so we use our GPS location as the originCamera.
+  m_arcGISArView->setOriginCamera(Camera());
+  m_arcGISArView->setTranslationFactor(1.0);
+
+  changeScene(true);
+}
+
+// Creates a scene based on the Streets base map.
+// Mode: Full-Scale AR
+void CppArExample::createStreetsScene()
+{
+  // create a scene with streets
+  m_scene = new Scene(BasemapType::Streets, this);
+  createSurfaceWithElevation();
+
+  // Set the location data source so we use our GPS location as the originCamera.
+  m_arcGISArView->setOriginCamera(Camera());
+  m_arcGISArView->setTranslationFactor(1.0);
+
+  changeScene(true);
+}
+
+// Creates a scene based on the ImageryWithLabels base map.
+// Mode: Full-Scale AR
+void CppArExample::createImageryScene()
+{
+  // create a scene with imagery
+  m_scene = new Scene(BasemapType::ImageryWithLabels, this);
+  createSurfaceWithElevation();
+
+  // Set the location data source so we use our GPS location as the originCamera.
+  m_arcGISArView->setOriginCamera(Camera());
+  m_arcGISArView->setTranslationFactor(1.0);
+
+  changeScene(true);
+}
+
+// Creates a test scene
+// Mode: Full-Scale AR
+void CppArExample::createFullScaleTestScene()
+{
+  // create scene
+  m_scene = new Scene(this);
+  createSurfaceWithElevation();
+
+  // create graphic overlay
+  auto* graphicsOverlay = new GraphicsOverlay(this);
+  Q_CHECK_PTR(m_sceneView);
+  m_sceneView->graphicsOverlays()->append(graphicsOverlay);
+
+  // create graphics
+  auto createSymbol = [graphicsOverlay](double x, double y, const QColor& color)
+  {
+    SimpleMarkerSceneSymbol* symbol = new SimpleMarkerSceneSymbol(
+          SimpleMarkerSceneSymbolStyle::Sphere, color, 0.1, 0.1, 0.1, SceneSymbolAnchorPosition::Center, graphicsOverlay);
+    graphicsOverlay->graphics()->append(new Graphic(Point(x, y, 0.00001), symbol, graphicsOverlay));
+  };
+
+  for (int i = -30; i <= 30; ++i)
+  {
+    for (int j = -30; j <= 30; ++j)
+    {
+      createSymbol(0.00001 * i, 0.00001 * j, Qt::blue);
```

Comment:
Implicit conversion here. For readability I would suggest this (here and below)

```suggestion
      createSymbol(0.00001 * i, 0.00001 * j, QColor(Qt::blue));
```

This is optional, the code is readable as it is.